lib/tasks/npm-task.js

```diff
@@ -165,31 +212,37 @@ class NpmTask extends Task {
     ui.writeLine(prependEmoji('🚧', 'Installing packages... This might take a couple of minutes.'));
     ui.startProgress(chalk.green(startMessage));
 
-    let promise;
-    if (this.useYarn) {
-      this.yarnVersion = result.yarnVersion;
-      let args = this.toYarnArgs(this.command, options);
-      promise = this.yarn(args);
-    } else {
-      let args = this.toNpmArgs(this.command, options);
-      promise = this.npm(args);
-
-      // as of 2018-10-09 npm 5 and 6 _break_ the hierarchy of `node_modules`
-      // after a `npm install foo` (deletes files/folders other than
-      // what was directly installed) in some circumstances, see:
-      //
-      // * https://github.com/npm/npm/issues/16853
-      // * https://github.com/npm/npm/issues/17379
-      //
-      // this ensures that we run a full `npm install` **after** any `npm
-      // install foo` runs to ensure that we have a fully functional
-      // node_modules hierarchy
-      if (result.npmVersion && semver.lt(result.npmVersion, '5.7.1')) {
-        promise = promise.then(() => this.npm(['install']));
+    try {
+      if (this.packageManager.name === 'yarn') {
+        let args = this.toYarnArgs(this.command, options);
+        await this.yarn(args);
+      } else if (this.packageManager.name === 'pnpm') {
+        let args = this.toPNPMArgs(this.command, options);
+        await this.pnpm(args);
+      } else {
+        let args = this.toNpmArgs(this.command, options);
+        await this.npm(args);
+
+        // as of 2018-10-09 npm 5 and 6 _break_ the hierarchy of `node_modules`
+        // after a `npm install foo` (deletes files/folders other than
+        // what was directly installed) in some circumstances, see:
+        //
+        // * https://github.com/npm/npm/issues/16853
+        // * https://github.com/npm/npm/issues/17379
+        //
+        // this ensures that we run a full `npm install` **after** any `npm
+        // install foo` runs to ensure that we have a fully functional
+        // node_modules hierarchy
+        let npmVersion = this.packageManager.version;
+        if (npmVersion && semver.lt(npmVersion, '5.7.1')) {
+          await this.npm(['install']);
+        }
```

Comment:
We can _probably_ remove this now

Reply:
probably, but it's unrelated to the pnpm stuff, so I'd rather not conflate the two things 😉